src/cli/cli.ts

```diff
@@ -122,24 +122,24 @@ program
 program
     .command('install [browserType...]')
     .description('ensure browsers necessary for this version of Playwright are installed')
-    .action(async function(args) {
+    .action(async function(args: any[]) {
       try {
         // Install default browsers when invoked without arguments.
         if (!args.length) {
           await registry.installBinaries();
           return;
         }
-        const browserNames: Set<BrowserName> = new Set(args.filter((browser: any) => allBrowserNames.has(browser)));
-        const browserChannels: Set<BrowserChannel> = new Set(args.filter((browser: any) => allBrowserChannels.has(browser)));
-        const faultyArguments: string[] = args.filter((browser: any) => !browserNames.has(browser) && !browserChannels.has(browser));
+        const binaries = args.map(arg => registry.findBinary(arg)).filter(b => !!b) as RegistryBinary[];
+        const browserChannels: Set<BrowserChannel> = new Set(args.filter(browser => allBrowserChannels.has(browser)));
+        const faultyArguments: string[] = args.filter((browser: any) => !binaries.find(b => b.name === browser) && !browserChannels.has(browser));
         if (faultyArguments.length) {
           console.log(`Invalid installation targets: ${faultyArguments.map(name => `'${name}'`).join(', ')}. Expecting one of: ${suggestedBrowsersToInstall}`);
           process.exit(1);
         }
-        if (browserNames.has('chromium') || browserChannels.has('chrome-beta') || browserChannels.has('chrome') || browserChannels.has('msedge') || browserChannels.has('msedge-beta'))
-          browserNames.add('ffmpeg');
-        if (browserNames.size)
-          await registry.installBinaries([...browserNames]);
+        if (browserChannels.has('chrome-beta') || browserChannels.has('chrome') || browserChannels.has('msedge') || browserChannels.has('msedge-beta'))
```

Comment:
We internally have somewhere already a reference that chrome-beta is chromium or that msedge is chromium afaik. So the best would be if we could use it there too to just check: if chromium -> ffmpeg.

Reply:
Yeah, this code will change one more time and will be simplified.